lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -19,6 +20,43 @@
 
 namespace json = llvm::json;
 
+using namespace circt;
+using namespace firrtl;
+
+/// Given a string \p target, starting with either '.' or '[', this function
+/// splits the string at every '[' and '.' and populates the \p annotations with
+/// the array of strings. Assumption, the \p target string is a well formed
+/// valid token specifying an instance of a bundle/array.
+void static parseSubFieldSubIndexAnnotations(StringRef target,
+                                             ArrayAttr &annotations,
+                                             MLIRContext *context) {
+  std::string temp = "";
```

Comment:
Please use `SmallString` instead of `std::string` to avoid allocations, and sink the declaration down past the early exits from the function.